src/tools/cephfs/JournalTool.cc

```diff
@@ -1056,12 +1067,16 @@ int JournalTool::erase_region(JournalScanner const &js, uint64_t const pos, uint
     return -EINVAL;
   }
 
-  // Serialize an ENoOp with the correct amount of padding
-  enoop = ENoOp(padding);
   bufferlist entry;
-  enoop.encode_with_header(entry, CEPH_FEATURES_SUPPORTED_DEFAULT);
+  if (type == "mdlog") {
+    // Serialize an ENoOp with the correct amount of padding
+    enoop = ENoOp(padding);
+    enoop.encode_with_header(entry, CEPH_FEATURES_SUPPORTED_DEFAULT);
+  } else if (type == "purge_queue") {
+    pi.pad_size = padding;
```

Comment:
@ukernel  yes, when calculating  pad_size,  all the space occupied by other fields are excluded.

Reply:
ENoOp has not action/ino/size, why the ways to calculate padding are the same

Reply:
because  the way to calculate padding is to exclude the length of a purgeitem whose pad_size is 0 which include actiob/ino/size